core/src/client/client.rs

```diff
@@ -828,11 +828,19 @@ impl TermInfo for Client {
     }
 
     fn term_common_params(&self, id: BlockId) -> Option<CommonParams> {
-        let block_number = self.last_term_finished_block_num(id).expect("The block of the parent hash should exist");
-        if block_number == 0 {
-            None
+        let state = self.state_at(id)?;
+        let metadata = state.metadata().unwrap().expect("Metadata always exist");
+
+        if let Some(term_params) = metadata.term_params() {
+            Some(term_params.clone())
         } else {
-            Some(self.common_params((block_number).into()).expect("Common params should exist"))
+            let block_number =
+                self.last_term_finished_block_num(id).expect("The block of the parent hash should exist");
+            if block_number == 0 {
+                None
+            } else {
+                Some(self.common_params((block_number).into()).expect("Common params should exist"))
+            }
```

Comment:
Could you add a commit description about why this change makes CodeChain read common params from the snapshot?